### DISCUSSION TITLE
How to configure to preprod env now testnet is deprecated?

### DISCUSSION THREAD

**Discussion by rpabani** [Python-Cardano/pycardano#83] [Q&A] [answered]
Hi

Thanks for creating this awesome package! Can you give any guidance on how to connect to the new preprod environment instead of testnet as its no longer supported?

https://book.world.dev.cardano.org/environments.html

Thanks

**Comment by cffls** [comment, marked as answer]
Hi there, 

You can use Blockfrost context to connect to preprod or preview. 
Example:

```python
from pycardano import BlockFrostChainContext, Network

chain_context = BlockFrostChainContext(
    project_id="YOUR_BLOCKFROST_PROJECT_ID",
    network=Network.TESTNET,
    base_url="https://cardano-preview.blockfrost.io/api",
)
```

APIs url of different networks could be found [here](https://docs.blockfrost.io/).

**Reply by rpabani** [reply]
I get an unexpected keyword argument 'base_url' TypeError when I try this. 

It looks like from the initialiser the base_url is set to None by default. 

`def __init__(
        self, project_id: str, network: Network = Network.TESTNET, base_url: str = None
    ):`

**Reply by cffls** [reply]
You will need to update the version of pycardano to `v0.6.1`.

**Reply by cffls** [reply]
Hi @rpabani , have you tried the solution in the post above? If it works, could you please kindly mark it as the answer. Thank you!

**Reply by rpabani** [reply]
Done. Thanks for your help
